northstar/src/runtime/island/mod.rs

```diff
@@ -0,0 +1,348 @@
+            .map_err(|e| Error::io("Failed to create duplex", e))?;
+
+        let (stdout, stderr, child_fd_map) = io::from_manifest(&container.manifest).await?;
+
+        let groups = groups(&container.manifest);
+
+        // Clone init
+        let flags = CloneFlags::CLONE_NEWPID | CloneFlags::CLONE_NEWNS;
+        match clone::clone(flags, Some(SIGCHLD as i32)) {
+            Ok(result) => match result {
+                unistd::ForkResult::Parent { child } => {
+                    let pid = child.as_raw() as Pid;
+                    let exit_status = Box::new(wait(container, pid, self.tx.clone()).await);
+
+                    Ok(IslandProcess::Created {
+                        pid,
+                        exit_status,
+                        intercom: intercom_runtime,
+                        io: (stdout, stderr),
+                    })
+                }
+                unistd::ForkResult::Child => init::init(
+                    &self.config,
+                    container,
+                    child_fd_map,
+                    &groups,
+                    intercom_init,
+                ),
+            },
+            Err(e) => panic!("Fork error: {}", e),
+        }
+    }
+}
+
+#[async_trait]
+impl Process for IslandProcess {
+    fn pid(&self) -> Pid {
+        match self {
+            IslandProcess::Created { pid, .. } => *pid,
+            IslandProcess::Started { pid, .. } => *pid,
+            IslandProcess::Stopped { pid } => *pid,
+        }
+    }
+
+    async fn start(mut self) -> Result<Self, Error> {
+        info!("Starting {}", self.pid());
+        match self {
+            IslandProcess::Created {
+                pid,
+                exit_status,
+                mut intercom,
+                io: _io,
+            } => {
+                // Signal the init process that it shall start the child (1)
+                intercom.send(LaunchProtocol::Go).ok();
+                // Wait for the notification from init that the child is launchend (8)
+                intercom.recv::<LaunchProtocol>().ok();
+
+                Ok(IslandProcess::Started {
+                    pid,
+                    exit_status,
+                    io: _io,
+                })
+            }
+            _ => unreachable!(),
+        }
+    }
+
+    /// Send a SIGTERM to the application. If the application does not terminate with a timeout
+    /// it is SIGKILLed.
+    async fn stop(
+        mut self,
+        timeout: time::Duration,
+    ) -> Result<(Self, ExitStatus), super::error::Error> {
+        let (pid, mut exit_status) = match self {
+            IslandProcess::Created {
+                pid, exit_status, ..
+            } => (pid, exit_status),
+            IslandProcess::Started {
+                pid,
+                exit_status,
+                io: _io,
+            } => (pid, exit_status),
+            IslandProcess::Stopped { .. } => unreachable!(),
```

Comment:
same thing for stop...shouldn't it just be a noop if the process is already stopped?

Reply:
Think this is a logical bug if someone calls stop on a stopped process.